blockchain/fullblocktests/generate.go

```diff
@@ -1089,6 +1092,95 @@ func Generate(includeLargeReorg bool) (tests [][]TestInstance, err error) {
 	})
 	rejected(blockchain.ErrFreshStakeMismatch)
 
+	// Attempt to add block with tickets voting on the wrong block
+	//
+	//   ... -> b36(8)
+	//                \-> bv6(9)
+	g.SetTip("b36")
+	g.NextBlock("bv6", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		// invalidate the vote by corrupting the output script
+		b.STransactions[0].TxOut[0].PkScript[2] &= 0x00
+	})
+	rejected(blockchain.ErrVotesOnWrongBlock)
+
+	// Attempt to add block with incorrect votebits set
+	// Everyone votes Yea, but block header says Nay
+	//
+	//   ... -> b36(8)
+	//                \-> bv7(9)
+	g.SetTip("b36")
+	g.NextBlock("bv7", outs[9], ticketOuts[9], func(b *wire.MsgBlock) {
+		// set the vote bit to Nay
```

Comment:
I'd personally leave these inner comments off.  The intro comment already describes it.  Also, the naming is inconsistent since it says "yea" while the variables are named yes.  I prefer the yes/no naming.